src/js/components/Player.js

```diff
@@ -0,0 +1,270 @@
+import React, { useState, useEffect } from "react";
+
+import { formatTime } from "../helpers/audio";
+
+import Meta from "./Meta";
+import PlayPause from "./PlayPause";
+import Range from "./Range";
+
+const Player = ({
+    url,
+    title,
+    artist,
+    theme = "aqoc-player",
+    features = {},
+}) => {
+    const includedFeatures = {
+        volume: true,
+        mute: true,
+        title: true,
+        ...features,
+    };
+    const [audio, setAudio] = useState(() => new Audio(url));
+
+    const [status, setStatus] = useState("pending"); // pending | loading | seeking | error | ready
+    const [playStatus, setPlayStatus] = useState("paused"); // playing | paused
+    const [times, setTimes] = useState({
+        duration: 0,
+        current: 0,
+    });
+    const [volume, setVolume] = useState(() => ({
+        value: 100, // range: 0-100
+        muted: false,
+    }));
+
+    const setupAudioListeners = () => {
+        setStatus("loading");
+        // audio.addEventListener("progress", handleProgress, false);
+        // audio.addEventListener("error", handleError, false);
+
+        // audio.addEventListener("abort", handleAbort, false);
+        // audio.addEventListener("audioprocess", handleAudioprocess, false);
+        audio.addEventListener("canplay", handleCanplay, false);
+        audio.addEventListener("canplaythrough", handleCanplaythrough, false);
+        // audio.addEventListener("complete", handleComplete, false);
+        audio.addEventListener("durationchange", handleDurationchange, false);
+        // audio.addEventListener("emptied", handleEmptied, false);
+        // audio.addEventListener("ended", handleEnded, false);
+        // audio.addEventListener("loadeddata", handleLoadeddata, false);
+        // audio.addEventListener("loadedmetadata", handleLoadedmetadata, false);
+        audio.addEventListener("pause", handlePause, false);
+        audio.addEventListener("play", handlePlay, false);
+        audio.addEventListener("playing", handlePlaying, false);
+        // audio.addEventListener("ratechange", handleRatechange, false);
+        // audio.addEventListener("seeked", handleSeeked, false);
+        // audio.addEventListener("seeking", handleSeeking, false);
+        // audio.addEventListener("stalled", handleStalled, false);
+        // audio.addEventListener("suspend", handleSuspend, false);
+        audio.addEventListener("timeupdate", handleTimeupdate, false);
+        // audio.addEventListener("volumechange", handleVolumechange, false);
+        // audio.addEventListener("waiting", handleWaiting, false);
+    };
+
+    // const handleAbort = (e) => {
```

Comment:
Do we need all this commented out code hanging around?

Reply:
Not "best practice" but the comments are really useful for me here (still in-progress on the enhanced player view).

Reply:
👍